### ISSUE TITLE
Cannot access static singletons in debugger.

### ISSUE THREAD

**Issue by lemmy101** [JetBrains/resharper-unity#566]
When using singleton pattern for MonoBehaviours (something I do a lot) it seems it cannot access any other singleton in the debugger, watch window or anywhere else unless within that class itself. e.g.

```
class Behaviour1 : MonoBehaviour
{
     public static Behaviour1 instance;
     public float value = 1;
     void Awake()
     {
          instance = this;
     }
}

class Behaviour2 : MonoBehaviour
{
    void Update()
    {
          float val = 20 * Behaviour1.instance.value;
           // debugging this function, Behaviour1.instance will not be visible to debugger. Impossible to determine 'Behaviour1.instance.value's value
           
    }
}
```

**Comment by IvanPashchenko** [comment]
Nope, not fixed. I have a repro, debugger team is looking into it.
<img width="489" alt="screen shot 2018-05-18 at 14 27 02" src="https://user-images.githubusercontent.com/8605238/40232513-c110d1a2-5aa7-11e8-9fa9-e2f839ec6448.png">

JetBrains Rider 2018.1 EAP
Build #RD-181.4952.220, built on May 18, 2018
Rider EAP User
Expiration date: June 17, 2018
JRE: 1.8.0_152-release-1136-b38 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.4

**Comment by IvanPashchenko** [comment]
Old runtime is required for repro.

**Comment by ArsenyChernyaev** [comment]
Should already be fixed in 18.2 EAP 6
